src/Evolution/Particles/MonteCarlo/EvolvePackets.cpp

```diff
@@ -37,7 +37,7 @@ void time_derivative_momentum_geodesic(
 }  // namespace detail
 
 void evolve_single_packet_on_geodesic(
-    const gsl::not_null<Packet*> packet, const double& time_step,
+    const gsl::not_null<Packet*> packet, double time_step,
```

Comment:
this should be `const double time_step`. For function arguments of type double, int, size_t that are constant, a function declaration in a header file should omit the `const`, but the definition should have the `const`